grazer/src/main/java/edu/asu/diging/grazer/core/domain/IFileMetadata.java

```diff
@@ -0,0 +1,40 @@
+package edu.asu.diging.grazer.core.domain;
+
+import java.util.Date;
+import java.util.List;
+
+import org.springframework.web.multipart.MultipartFile;
+
+/**
+ * This class stores all the details about a transformation file and pattern file 
+ * i.e their IDs, labels, descriptions, file names, data uploaded, and uploader.
+ * 
+ * @author mshah18
+ *
+ */
+public interface IFileMetadata {
```

Comment:
You say this file "stores all the details about a transformation file and pattern file", so it should be named ITransformationMetadata or ITransformationFilesMetadata. Your filename implies that is is for metadata about one file.